### PR TITLE
Change `Cast`/`Forge` to `snforge` and `sncast`

### LINKED ISSUE #998: Consider adding `Sn` prefix to project names in all places
Follow-up of #958 

Our project binaries use `sn` prefixes in all places, but usages of names with and without prefixes in other places are inconsistent. 

E.g. https://github.com/foundry-rs/starknet-foundry/blob/e0bdc7ce715bb2112380c7be8ddc6503e45ee494/crates/cast/src/main.rs#L21
or in docs headings

We should consider if we want to use consistent names in all places.

### PR DESCRIPTION
**Pull request by Arcticae**
Closes #998 

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/appendix/snforge.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Forge CLI Reference
+
+* [`snforge test`](./snforge/test.md)
+* [`snforge init`](./snforge/init.md)
+* [`snforge clean-cache`](./snforge/clean-cache.md)
+
+You can check your version of Forge via `snforge --version`.
```
Comment (THenry14) [inline docs/src/appendix/snforge.md:7]:
```suggestion
You can check your version of snforge via `snforge --version`.
```
? I'm not sure

File: docs/src/SUMMARY.md [hunk 2]
```diff
@@ -39,10 +39,10 @@
 
 # Appendix
 
-* [Forge Commands](appendix/forge.md)
-    * [test](appendix/forge/test.md)
-    * [init](appendix/forge/init.md)
-    * [clean-cache](appendix/forge/clean-cache.md)
+* [`snforge` Commands](appendix/forge.md)
```
Comment (THenry14) [inline docs/src/SUMMARY.md:42]:
```suggestion
* [`snforge` Commands](appendix/snforge.md)
```
this should be probably changed as well?

Comment (THenry14) [reply docs/src/SUMMARY.md:42]: there are `forge.md` and `snfroge.md` files if I see correctly in the src/appendix. is this intended?

File: docs/src/appendix/snforge.md (hunk 1 above)
Comment (THenry14) [reply docs/src/appendix/snforge.md:7]: the same goes for multiple places here (like appendix/snforge-library/print.md etc), how should we approach it? I guess it should also be changed? (there aren't any occurences of cast | Cast, that's why I wonder)

(to see where forge | Forge is still used I used: )
```
starknet-foundry/docs/src $ ➜ grep -rnw -e "forge"
    and
starknet-foundry/docs/src $ ➜ grep -rnw -e "Forge"
```

File: docs/src/SUMMARY.md (hunk 2 above)
Comment (Arcticae) [reply docs/src/SUMMARY.md]: the other one was not needed

File: docs/src/appendix/snforge.md (hunk 1 above)
Comment (Arcticae) [reply docs/src/appendix/snforge.md]: thanks for the tip @THenry14 <3

File: README.md [hunk 3]
```diff
@@ -25,9 +25,9 @@ Need help getting started with Starknet Foundry? Read the
 
 Starknet Foundry, like its [Ethereum counterpart](https://github.com/foundry-rs/foundry), consists of different modules
 
-- [Forge](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/forge): Starknet testing
+- [snforge](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/forge): Starknet testing
```
Comment (cptartur) [inline README.md:28]:
```suggestion
- [`snforge`](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/forge): Starknet testing
```

How about that?

File: README.md [hunk 4]
```diff
@@ -25,9 +25,9 @@ Need help getting started with Starknet Foundry? Read the
...
   framework (like Truffle, Hardhat and DappTools but for Starknet).
-- [Cast](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/sncast): All-in-one tool for
+- [sncast](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/sncast): All-in-one tool for
```
Comment (cptartur) [inline README.md:30]:
```suggestion
- [`sncast`](https://github.com/foundry-rs/starknet-foundry/tree/master/crates/sncast): All-in-one tool for
```

**Review by cptartur — APPROVED** [review]

**Review by THenry14 — APPROVED** [review]
💪

**Pull request merged**